app/controllers/concerns/saml_idp_auth_concern.rb

```diff
@@ -6,13 +6,20 @@ module SamlIdpAuthConcern
     # rubocop:disable Rails/LexicallyScopedActionFilter
     before_action :validate_saml_request, only: :auth
     before_action :validate_service_provider_and_authn_context, only: :auth
+    before_action :sign_out_if_forceauthn_is_true_and_user_is_signed_in, only: :auth
```

Comment:
Note this placement in the chain is _very_ important - we need to sign the user out _before_ we store the SAML request or we lose the entire request context.

Reply:
if the order is that important, maybe record that with a comment in the source code?

Reply:
Good idea